libr/debug/p/native/linux/linux_coredump.c

```diff
@@ -813,9 +812,7 @@ static proc_per_process_t *get_proc_process_content(RDebug *dbg) {
 		long unsigned int no_lui;
 		long int no_li;
 		int no_num;
-		sscanf (buff, "%d %s %c %d %d %d %d %d %u %lu %lu %lu %lu"
-				"%lu %lu %ld %ld %ld %ld %ld",
-			&p->pid, no_str, &p->s_name, &p->ppid, &p->pgrp, &no_num,
+		r_str_scanf (buff, "%d %.s %c %d %d %d %d %d %u %lu %lu %lu %lu %lu %lu %ld %ld %ld %ld %ld", &p->pid, sizeof (no_str), no_str, &p->s_name, &p->ppid, &p->pgrp, &no_num,
```

Comment:
i added a test for this scanf expression and find out the parsing of %c was wrong in the original bscanf code. fixed it and added some more tests. hope this work now for your pr :)

https://github.com/radareorg/radare2/pull/22578

additionally, you should check for the return value for this call, (which should return 20).

Also all those no_lui, no_num, no_li... can be removed if we just use %*d or %*lu, the `*` is the supression modifier, which may make the number of arguments much shorter in this case.